pkg/reconcile/pipeline/context/impl.go

```diff
@@ -118,15 +121,43 @@ func (i *impl) Services() ([]pipeline.Service, error) {
 func (i *impl) Applications() ([]pipeline.Application, error) {
 	if i.applications == nil {
 		if ref := i.serviceBinding.Spec.Application; ref != nil {
+			if i.serviceBinding.Spec.Application.Name != "" &&
+				i.serviceBinding.Spec.Application.LabelSelector != nil &&
+				i.serviceBinding.Spec.Application.LabelSelector.MatchLabels != nil {
+				return nil, errs.New("Both label selector and name for application should not be used")
```

Comment:
we can extract error into a var that can be used later in test assertion - instead of string copying.

Reply:
Created separate issue: #974